### PR TITLE
Update evaluation for unify coco and voc

### PR DESCRIPTION
**Pull request by yghstill**
1.unify coco and voc evaluation
2.support compute per-category AP/Precision/Recall and draw P-R Curve.

```
Evaluate annotation type *bbox*
DONE (t=51.75s).
Accumulating evaluation results...
DONE (t=7.52s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.379
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.596
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.406
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.224
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.414
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.489
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.313
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.492
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.517
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.326
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.556
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.650
2020-11-23 06:30:43,870-INFO: Per-category AP
+---------------+-------+--------------+-------+----------------+-------+
| category      | AP    | category     | AP    | category       | AP    |
+---------------+-------+--------------+-------+----------------+-------+
| person        | 0.523 | bicycle      | 0.286 | car            | 0.417 |
| motorcycle    | 0.396 | airplane     | 0.598 | bus            | 0.601 |
| train         | 0.578 | truck        | 0.309 | boat           | 0.258 |
| traffic light | 0.270 | fire hydrant | 0.614 | stop sign      | 0.627 |
| parking meter | 0.458 | bench        | 0.222 | bird           | 0.336 |
| cat           | 0.598 | dog          | 0.551 | horse          | 0.526 |
| sheep         | 0.490 | cow          | 0.514 | elephant       | 0.602 |
| bear          | 0.632 | zebra        | 0.632 | giraffe        | 0.625 |
| backpack      | 0.153 | umbrella     | 0.347 | handbag        | 0.124 |
| tie           | 0.303 | suitcase     | 0.351 | frisbee        | 0.620 |
| skis          | 0.205 | snowboard    | 0.331 | sports ball    | 0.461 |
| kite          | 0.396 | baseball bat | 0.249 | baseball glove | 0.341 |
| skateboard    | 0.479 | surfboard    | 0.343 | tennis racket  | 0.430 |
| bottle        | 0.378 | wine glass   | 0.332 | cup            | 0.396 |
| fork          | 0.290 | knife        | 0.151 | spoon          | 0.130 |
| bowl          | 0.406 | banana       | 0.217 | apple          | 0.185 |
| sandwich      | 0.292 | orange       | 0.294 | broccoli       | 0.206 |
| carrot        | 0.174 | hot dog      | 0.300 | pizza          | 0.486 |
| donut         | 0.425 | cake         | 0.336 | chair          | 0.246 |
| couch         | 0.365 | potted plant | 0.239 | bed            | 0.369 |
| dining table  | 0.245 | toilet       | 0.563 | tv             | 0.526 |
| laptop        | 0.558 | mouse        | 0.577 | remote         | 0.275 |
| keyboard      | 0.470 | cell phone   | 0.319 | microwave      | 0.530 |
| oven          | 0.296 | toaster      | 0.295 | sink           | 0.346 |
| refrigerator  | 0.497 | book         | 0.129 | clock          | 0.493 |
| vase          | 0.344 | scissors     | 0.232 | teddy bear     | 0.419 |
| hair drier    | 0.004 | toothbrush   | 0.177 | None           | None  |
+---------------+-------+--------------+-------+----------------+-------+
2020-11-23 06:30:43,871-INFO: per-category PR curve has output to bbox_pr_curve folder.
```

### CODE REVIEW

**Review by wangxinxin08 — APPROVED** [review]
LGTM

File: ppdet/evaluation/result_out.py [hunk 1]
```diff
@@ -0,0 +1,338 @@
+# Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
```
Comment (qingqing01) [inline ppdet/evaluation/result_out.py:1]: result_out.py -> post_processing.py or transform_prediction.py, ...

File: ppdet/data/source/coco.py [hunk 2]
```diff
@@ -14,9 +14,18 @@
...
+import json
+import itertools
 
 from .dataset import DataSet
 from ppdet.core.workspace import register, serializable
+from ppdet.evaluation.map_utils import draw_pr_curve
+from ppdet.evaluation.result_out import proposal2out, bbox2out, mask2out, segm2out
+
+from pycocotools.coco import COCO
+from pycocotools.cocoeval import COCOeval
```
Comment (qingqing01) [inline ppdet/data/source/coco.py:28]: 把cocoapi importing放到函数里，这样只有VOC数据时，不安装cocoapi,  也可以训练, 或者只做推理也可以

File: ppdet/data/source/coco.py [hunk 3]
```diff
@@ -14,9 +14,18 @@
 
 import os
 import numpy as np
+import os
```
Comment (qingqing01) [inline ppdet/data/source/coco.py:17]: os重复import

**Pull request closed without merging**